### PR TITLE
Add restriction to handle user behavior of adding future applications

### LINKED ISSUE #169: [PE-D][Tester C] Adding an application with a future date should not be allowed
Currently, the application allows the user to add an application with a future date. However, it does not make sense for a user to be able to apply for an internship of a future date

Example: Today's date is 28/10/22. However, I am allowed to add an internship application where the date I've applied is in the year 2024. 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: maxtance/ped#9

### PR DESCRIPTION
**Pull request by eesung00**
Resolved #169 

Added one application exceptions to restrict user from adding applications with future dates

Todo:
- [x] Add new test case to test the restriction

### CODE REVIEW

**Review by jhchee18 — COMMENTED** [review]
imo, the validation check should be in `Date` class instead of `Application` class. It seems abit out of place when there is a sudden date check in application. What do you think? @eesung00

**Comment by eesung00** [comment]
> imo, the validation check should be in `Date` class instead of `Application` class. It seems abit out of place when there is a sudden date check in application. What do you think? @eesung00

On a second thought, I think putting the checking in the `Application` is better? Because its something related to Application instead of the Date object creation.

Just like how the interview is checking its date is after application date in `Application` class also? 🤔

File: src/main/java/seedu/application/model/application/Application.java [hunk 1]
```diff
@@ -171,6 +174,12 @@ public boolean isSameApplication(Application otherApplication) {
                 && otherApplication.position.equals(position);
     }
 
+    private void applicationDateIsInFuture(Date date) {
+        if (date.value.isAfter(LocalDate.now())) {
```
Comment (jhchee18) [inline src/main/java/seedu/application/model/application/Application.java]:
```suggestion
        if (date.isAfter(LocalDate.now())) {
```
can define a `isAfter()` method in `Date` class to adhere to SLAP

**Review by jhchee18 — COMMENTED** [review]
Sounds reasonable. Just one code quality improvement recommended.

**Review by jolynloh — COMMENTED** [review]
LGTM!

**Pull request merged**